sbi/neural_nets/embedding_nets/causal_cnn.py

```diff
@@ -0,0 +1,176 @@
+    padding_size = dilation * (kernel_size - 1)
+    padding = nn.ZeroPad1d(padding=(padding_size, 0))
+    conv_layer = nn.Conv1d(
+        in_channels=in_channels,
+        out_channels=out_channels,
+        kernel_size=kernel_size,
+        dilation=dilation,
+        stride=1,
+        padding=0,
+    )
+    return nn.Sequential(padding, conv_layer, activation)
+
+
+class CausalCNNEmbedding(nn.Module):
+    def __init__(
+        self,
+        input_shape: Tuple,
+        in_channels: int = 1,
+        out_channels_per_layer: Optional[List] = None,
+        dilation_per_layer: Optional[List] = None,
+        num_conv_layers: int = 5,
+        activation: nn.Module = nn.LeakyReLU(inplace=True),
+        pool_kernel_size: int = 160,
+        kernel_size: int = 2,
+        aggregator: Optional[nn.Module] = None,
+        output_dim: int = 20,
+    ):
+        """Embedding network that uses 1D causal convolutions
+        This is a simplified version of the architecture introduced for
+        the speech recognition task in the WaveNet paper (van den Oord, et al. (2016))
+
+        After several dilated causal convolutions (that maintain the dimensionality
+        of the input), an aggregator network is used to bring down the dimensionality.
+        You can provide an aggregator network that you deem reasonable for your data
+
+        Args:
+            input_shape: Dimensionality of the input e.g. (num_timepoints,),
+                    currently only 1D is supported.
+            in_channels: Number of input channels, default = 1.
+            out_channels_per_layer: number of out_channels for each layer, number
+                    of entries should correspond with num_conv_layers passed below.
+                    Default = 16 in every convolutional layer.
+            dilation_per_layer: dilation size per layer, by default the cyclic,
+                    exponential scheme from WaveNet is used.
+            num_conv_layers: the number of causal convolutional layers
+            kernel_size: size of the kernels in the causal convolutional layers.
+            activation: activation function to use between convolutions,
+                    default = LeakyReLU.
+            pool_kernel_size: pool size to use for the AvgPool1d operation after
+                    the causal convolutional layers.
+            aggregator: aggregation net that reduces the dimensionality of the data
+                    to a low-dimensional embedding.
+            output_dim: number of output units in the final layer when using
+                    the default aggregation
+
+        """
+
+        super(CausalCNNEmbedding, self).__init__()
+        assert isinstance(input_shape, Tuple), (
+            "input_shape must be a Tuple of size 1, e.g. (timepoints,)."
+        )
+        assert len(input_shape) == 1, "Currently only 1D causal CNNs are supported."
+        self.input_shape = (in_channels, *input_shape)
+
+        tot_timepoints = input_shape[0]
```

Comment:
it's fine to be more explicit as long as the variable names do not become too long. so I would go with `total_timepoints` or `num_timepoints`